Open twitter tapping on tweet

maxxx777 opened the pull request:
The PR closes #129 issue.
`selectMessageTrigger` was added to `MessagesTableViewController` to handle select row in table view.
`LinkResponsiveTextView` was introduced to let open links as by selecting a table view cell as selecting a link in table view cell.
No UI changes.

### Screenshots (before the changes)
![IMG_0007](https://user-images.githubusercontent.com/2142832/66160813-39d8c300-e62b-11e9-8343-0357a82bc997.PNG)
![IMG_0008](https://user-images.githubusercontent.com/2142832/66160814-39d8c300-e62b-11e9-9071-e7d6aaf73964.PNG)

### Screenshots (after the changes)
![IMG_0010](https://user-images.githubusercontent.com/2142832/66160835-478e4880-e62b-11e9-8e15-92794a168f75.PNG)
![IMG_0009](https://user-images.githubusercontent.com/2142832/66160832-478e4880-e62b-11e9-9b0e-04369ea8a929.PNG)

## Checklist

### Before merging the PR

- [x] If applicable, did you add a before / after screenshot of the feature?
- [x] Did you update the [CHANGELOG.md](https://github.com/criticalmaps/criticalmaps-ios/blob/master/CHANGELOG.md)?

mltbnz commented on CriticalMass/TwitterViewController.swift:
```diff
@@ -50,3 +53,19 @@ class TwitterViewController: UIViewController {
+private extension TwitterViewController {
+    func openTweet(_ tweet: Tweet) {
+        if let webURL = tweet.webURL, UIApplication.shared.canOpenURL(webURL) {
+            UIApplication.shared.open(webURL, options: [:], completionHandler: nil)
```
Could we also add the universal link variant of the user has installed the Twitter app? Not entirely sure if that even works with the tweet link but would be great. Then we could route to the app if `canOpenUrl` is true and if not we open the Link in Safari or if it was for me could also be SafariViewController. 

This is also fine but maybe you could check if twitter app link is possible

mltbnz replied:
I rechecked on my phone and it already works like a charm 🦢
Let's merge (once the conflict is resolved)

maxxx777 replied:
the web link should open the Twitter app if It's installed. So no need to add the twitter universal link.

maxxx777 commented:
@lennet @mltbnz thanks for the comments. will fix conflicts for this and other PRs later this month when be back after vacations :)

maxxx777 commented:
- Fixed the conflicts 
- Added highlighting when cell selected

![Simulator Screen Shot - iPhone 8 Plus - 2019-10-17 at 15 17 25](https://user-images.githubusercontent.com/2142832/67012317-7a3b4500-f0f1-11e9-86ac-169420b8733f.png)
![Simulator Screen Shot - iPhone 8 Plus - 2019-10-17 at 15 17 48](https://user-images.githubusercontent.com/2142832/67012318-7a3b4500-f0f1-11e9-987d-254ec32ee804.png)

lennet commented on CriticalMass/TweetTableViewCell.swift:
```diff
@@ -68,6 +68,12 @@ class TweetTableViewCell: UITableViewCell, MessageConfigurable, IBConstructable
+        if #available(iOS 13.0, *) {
+            // workaround for iOS 13
+            // set background color to clear to keep the same color for background text view and background cell view
+            // when the cell is highlighted
```
is the textview backgroundcolor on pre iOS 13 clear by default?

maxxx777 replied:
I set `Default` color in storyboard. It looks like it `white` in pre iOS 13 (`UIExtendedGrayColorSpace 1 1`). 
But if I set `white`,  it stays `white` in iOS 13 when cell selected:

![Simulator Screen Shot - iPhone 8 Plus - 2019-10-17 at 16 08 22](https://user-images.githubusercontent.com/2142832/67016433-79f27800-f0f8-11e9-88bf-c002c7110a18.png)
